clang/include/clang/Basic/TokenKinds.def

```diff
@@ -658,6 +658,9 @@ KEYWORD(out                         , KEYHLSL)
 #define HLSL_INTANGIBLE_TYPE(Name, Id, SingletonId) KEYWORD(Name, KEYHLSL)
 #include "clang/Basic/HLSLIntangibleTypes.def"
 
+// HLSL Type traits.
+TYPE_TRAIT_2(__is_scalarized_layout_compatible, IsScalarizedLayoutCompatible, KEYHLSL)
```

Comment:
Can you prefix this with `__builtin_`? We have bad experiences with accidental naming conflicts (not that I think this will run into it), and we're starting to add a prefix now to help avoid that.